server/src/main/java/eu/fasten/server/plugins/kafka/FastenKafkaPlugin.java

```diff
@@ -220,28 +222,40 @@ public void handleConsuming() {
         // processed
         sendHeartBeat(connNorm);
 
+        // normally, we ONLY wait on PRIO and ONLY when there have been no messages in any lane...
+        var prioTimeout = hadMessagesOnLastPollCycle ? Duration.ZERO : POLL_TIMEOUT;
```

Comment:
@proksch 
Why poll timeouts are different? Could you please elaborate a bit?

Reply:
The idea is that -in general- we do not wait as long as there are messages to process (Duration.ZERO), but when there have been no messages whatsoever in the last cycle, neither on NORMAL nor on PRIO, we wait a bit... otherwise, we would end up sending constant requests to Kafka.

Reply:
If you would always wait for a fixed wait time (as it was before), then Lane A will slow down Lane B when there are no messages on A, but messages on B (You can use any combination of NORMAL/PRIO for this example).